### PR TITLE
Project UMD 

### PR DESCRIPTION
**Pull request by Shaderpixel**
- Integrate UMD wrapping of JS module files
- Integrate Browserify for bundling module files in example theme
- Update documentation with new processes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,35 +7,42 @@ Digital Campus Framework
...
 * The `assets/src` directory contains the core scripts and files for DCF.
-* The `assets/build` directory contains temporary files between build processes (e.g. concatenated JS files before minification).
+* The `assets/build` directory contains temporary files in-between build processes (e.g. transpiled JS files before 
+minification).
 * The `assets/dist` directory contains production ready scripts and files for DCF.
-* The `example` directory contains compiled CSS and an `index.shtml` to preview the theme. 
+* The `example` directory contains an example setup of how to use the core styles and JS modules. SASS files are 
+compiled to CSS and the JS module files are bundled, transpiled, and minified using Browserify. The end product
+can be seen live at [digitalcampus.us](http://www.digitalcampus.us/).  
+//TODO: Ask Michael if people can look at the index.shtml without doing the whole symlink process
```
Comment (Shaderpixel) [inline README.md]: @mfairchild365 we need to visit about this

File: assets/build/js/app/postBabel/common/dcf-dialog.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+'use strict';
```
Comment (mfairchild365) [inline assets/build/js/app/postBabel/common/dcf-dialog.js]: This appears to be a temporary build file. I would avoid committing artifacts like this to git. Perhaps add remove these and then add a line to the `.gitignore` file?

File: build-utils/build-paths.js [hunk 3]
```diff
@@ -16,7 +19,8 @@ const picturefill = require.resolve('picturefill');
 const objectFitImages = require.resolve('object-fit-images');
 const detailsPolyfill = require.resolve('details-polyfill');
 
-console.log([`${mustardJsSrcPath}/**/*.js`, `${picturefill}`, `${objectFitImages}`, detailsPolyfill]);
+// console.log([`${mustardJsSrcPath}/**/*.js`, `${picturefill}`, `${objectFitImages}`, detailsPolyfill]);
+// TODO check to see if all the path properties are still being used
```
Comment (mfairchild365) [inline build-utils/build-paths.js]: Remove these comments

File: build-utils/dist-paths.js [hunk 4]
```diff
@@ -22,8 +22,11 @@ module.exports = {
 	cssDest: path.join(commonPaths.outputDist, 'css'),
 	optionalAppGlob: `${optionalAppPath}/**/*.js`,
 	optionalAppDest: path.join(commonPaths.outputDist,'js', 'app', 'optional'),
-	commonAppSrc: path.join(buildPaths.commonAppDest, buildNames.appJs),
+	// commonAppSrc: path.join(buildPaths.commonAppDest, buildNames.appJs),
```
Comment (mfairchild365) [inline build-utils/dist-paths.js]: Remove debug comment

File: build-utils/example-theme-gulp-tasks.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+/* ----------------- */
+/* EXAMPLE STYLE LINT TASKS
+/* ----------------- */
+function stylelint() {
+	$.fancyLog('----> //** Linting Example SCSS files');
+	return $.pump([
+		gulp.src(buildPaths.exampleScssGlob),
+		customPlumber('Error Running stylelint:example:cached'),
+		// $.debug({title: 'All Files - [stylelint:newer:Example]'}), // uncomment to see src files
```
Comment (mfairchild365) [inline build-utils/example-theme-gulp-tasks.js]: Remove debug comments here too

File: build-utils/uglify.js [hunk 6]
```diff
@@ -17,9 +17,9 @@ function uglifyNewer (src, dest, taskName, newerDest) {
 			gulp.src(src),
 			customPlumber(`Error Running ${taskName} task`),
 			$.sourcemaps.init({loadMaps:true}),
-			$.debug({title: 'All Files'}),
+			// $.debug({title: 'All Files'}),
```
Comment (mfairchild365) [inline build-utils/uglify.js]: And remove debug comments here too

File: build-utils/umd-tasks-optional.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+const buildPaths = require('./build-paths');
+const customPlumber = require('./custom-plumber');
+
+/**
+ * Warning: if no exports option set, the exported variable (for commonJS) is taken from the file name and capitalized
+ * so if you are returning something, make sure that you either specify an exports option or rename the file to
+ * match. See https://github.com/eduardolundgren/gulp-umd#options
+ */
+
+// const test = [
```
Comment (mfairchild365) [inline build-utils/umd-tasks-optional.js]: Remove commented out test code

File: gulpfile.babel.js [hunk 8]
```diff
@@ -41,7 +43,7 @@ gulp.task('stylelint:newer', (done) => {
 		customPlumber('Error Running stylelint:newer'),
 		// $.debug({title: 'All Files - [stylelint:newer]'}), // uncomment to see src files
 		$.newer({dest: distPaths.scssDest}),
-		$.debug({title: 'Passed Through - [stylelint:newer]'}), // uncomment to see files passed through
+		// $.debug({title: 'Passed Through - [stylelint:newer]'}), // uncomment to see files passed through
```
Comment (mfairchild365) [inline gulpfile.babel.js]: More unnecessary comments

File: build-utils/build-paths.js (hunk 3 above)
Comment (Shaderpixel) [reply build-utils/build-paths.js]: Removed.

File: build-utils/dist-paths.js (hunk 4 above)
Comment (Shaderpixel) [reply build-utils/dist-paths.js]: Removed unused path variables.

File: build-utils/example-theme-gulp-tasks.js (hunk 5 above)
Comment (Shaderpixel) [reply build-utils/example-theme-gulp-tasks.js]: Removed.

File: build-utils/uglify.js (hunk 6 above)
Comment (Shaderpixel) [reply build-utils/uglify.js]: Removed.

File: build-utils/umd-tasks-optional.js (hunk 7 above)
Comment (Shaderpixel) [reply build-utils/umd-tasks-optional.js]: Removed.

File: assets/build/js/app/postBabel/common/dcf-dialog.js (hunk 2 above)
Comment (Shaderpixel) [reply assets/build/js/app/postBabel/common/dcf-dialog.js]: Build file will be excluded from distribution, pre-transpiled and pre-minified module files are now included inside of dist

File: gulpfile.babel.js (hunk 8 above)
Comment (Shaderpixel) [reply gulpfile.babel.js]: 🗑

**Pull request merged**